noir-projects/noir-contracts/contracts/easy_private_voting_contract/src/main.nr

```diff
@@ -21,14 +21,18 @@ contract EasyPrivateVoting {
     fn constructor(admin: AztecAddress) {
         storage.admin.write(admin);
         storage.vote_ended.write(false);
+        storage.active_at_block.initialize(context.block_number() as u32);
     }
     // docs:end:constructor
 
     // docs:start:cast_vote
     #[aztec(private)] // annotation to mark function as private and expose private context
     fn cast_vote(candidate: Field) {
-        let msg_sender_npk_m_hash = get_npk_m_hash(&mut context, context.msg_sender());
-        // TODO (#6312): This will break with key rotation. Fix this. Can vote multiple times by rotating keys.
+        // Below, we make sure to get our nullifier public key at a specific block. By pinning the nullifier public key at a specific block,
```

Comment:
![Winnie The Pooh Happy Dance GIF by Leon Denise](https://media2.giphy.com/media/l3vRlT2k2L35Cnn5C/giphy.gif)